### DISCUSSION TITLE
Middle button only drag

### DISCUSSION THREAD

**Discussion by ishtanzar** [Leaflet/Leaflet#8551] [Q&A]
Hi,
I would like to keep map panning but only with the middle mouse button so the left drag could be used for another feature (custom distance calculation).
Seems like left & middle button for dragging is hard-coded in the core.
Thanks

**Comment by zxh19890103** [comment]
I've a solution, but it's dangerous. you can just give a consideration.

https://github.com/Leaflet/Leaflet/blob/main/src/dom/Draggable.js#L49

it's the method `enable` of `Draggable`.

```js
enable: function () {
                 if (this._enabled) { return; }
		DomEvent.on(this._dragStartTarget, START, this._onDown, this);
		this._enabled = true;
},
```

where the `START` equals to `"mousedown"`, which makes `left and middle` button both lead to map dragging.

so you can just rewrite it, to exclude the `left` button listening.

```js
enable: function () {
                 if (this._enabled) { return; }
		// DomEvent.on(this._dragStartTarget, START, this._onDown, this);
                DomEvent.on(this._dragStartTarget, START, (e) => {
                  if (!isMiddleButtonPressed(e)) return;
                  this._onDown(e);
                }, this);
		this._enabled = true;
},
```
